packages/shared/src/convert-to-className.js

```diff
@@ -58,7 +58,9 @@ export function convertStyleToClassName(
 
   // NOTE: '<>' is used to keep existing hashes stable.
   // This should be removed in a future version.
-  const className = classNamePrefix + createHash('<>' + stringToHash);
+  const className = dev
+    ? `${classNamePrefix}${dashedKey}_${createHash('<>' + stringToHash)}`
```

Comment:
Let's put a separator in dev for this:

```suggestion
    ? `${classNamePrefix}-${dashedKey}_${createHash('<>' + stringToHash)}`
```

OR:

```suggestion
    ? `${dashedKey}-${classNamePrefix}${createHash('<>' + stringToHash)}`
```

Reply:
There might also be some value in adding pseudo class or `_media` as prefixes, but let's save that for a follow-up.

Reply:
Let's make it follow this pattern: `${key}-${classNamePrefix}${hash}`

1. We keep the prefix+hashed part the same, whether it's dev or prod.
2. The key matches the authored key name.
3. Simpler code